File: AMS_电子对账/code/server/source/com.yss.uco.elecreco/src/main/java/com/yss/uco/elecreco/er/erresult/service/IErResultService.java
package com.yss.uco.elecreco.er.erresult.service;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import com.yss.framework.api.mvc.biz.IServiceBus;
import com.yss.framework.api.restful.annotations.RestfulSupported;
import com.yss.framework.api.servlet.pojo.QueryRes;

@RestfulSupported
public interface IErResultService extends IServiceBus {
	
	public QueryRes queryOrigDataByCondition(HashMap<String, Object> paraMap);
	public boolean isExistAssetCode(String assetCode);
	

	/**
	 * 判断指标是否一致
	 * @param csn 报文序号
	 * @param rowFlag 指标代码或者科目代码
	 * @return
	 */
	public boolean isSameData(String csn, List<String> rowFlag);
	
	public int queryUnAcceptCount(String csn);
	
	/**
	 * 获取组合关联机构设置的托管人
	 * @param portcodes
	 * @return
	 */
	public Map<String, String> getPortRelaOrgData(String portcodes);

}

File: AMS_电子对账/code/source/YssElecReco/Service/Er/IErResultService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using FAST.Common.Service.Pojo;
using FAST.Common.Service.Services.Base;
using FAST.Common.Service.Attributes;
using YssElecReco.Context;

namespace YssElecReco.Service.Er
{
    /// <summary>
    /// 产生电子对账核对数据service
    /// chenyoulong 20121213
    /// v1.0.0.6
    /// </summary>
    [ServiceAttribute(ServiceId = ElecrecoServiceIdConstant.OSGI_ELECRECO, ServiceController = "com.yss.uco.elecreco.er.erresult.controller.IErResultServiceController")]
    public interface IErResultService : IServiceBus
    {
        /// <summary>
        /// 获取未核对前的原始数据
        /// </summary>
        /// <param name="paraDict">paraDict</param>
        /// <returns>QueryRes</returns>
        [MethodAttribute]
        QueryRes queryOrigDataByCondition(Dictionary<string, string> paraDict);

        Dictionary<string, string> getPortRelaOrgData(String portcodes);
    }
}
